rioxarray/raster_writer.py

```diff
@@ -155,7 +155,9 @@ def to_raster(self, xarray_dataarray, tags, windowed, lock, compute, **kwargs):
         **kwargs
             Keyword arguments to pass into writing the raster.
         """
-        dtype = kwargs["dtype"]
+        numpy_dtype = kwargs["dtype"]
+        if xarray_dataarray.encoding.get("_rasterio_dtype") == "complex_int16":
+            kwargs["dtype"] = "complex_int16"
```

Comment:
> we will need to have a way to get the rasterio and numpy dtype. For the writer, I think that it is best to assume that the "dtype" kwarg could be either the rasterio or numpy dtype and automatically determine the numpy/rasterio dtype from it. 

i think the only special dtype is `complex_int16` so i went with this approach, let me know if you prefer something else. tests pass except for the new one (requires ` pip install git+https://github.com/mapbox/rasterio.git@maint-1.2`, fixes aren't yet in the master branch

Reply:
I think the fixes are in master now.